### PR TITLE
integration: add more lease tests

### PR DESCRIPTION
**Pull request by gyuho**
Fix https://github.com/coreos/etcd/issues/6102.

### CODE REVIEW

**Comment by xiang90** [comment]
Can you try to run these test before commit https://github.com/coreos/etcd/pull/6098?

File: integration/v3_lease_test.go [hunk 1]
```diff
@@ -366,6 +366,242 @@ func TestV3LeaseFailover(t *testing.T) {
...
 
+// TestV3LeaseRecoverAndRevoke ensures that revoking a lease after restart deletes the attached key.
+func TestV3LeaseRecoverAndRevoke(t *testing.T) {
+	clus := NewClusterV3(t, &ClusterConfig{Size: 1})
+	defer clus.Terminate(t)
+
+	kvc := toGRPC(clus.Client(0)).KV
+	lsc := toGRPC(clus.Client(0)).Lease
+
+	lresp, err := lsc.LeaseGrant(context.TODO(), &pb.LeaseGrantRequest{TTL: 5 * 60})
```
Comment (xiang90) [inline integration/v3_lease_test.go]: define the 5 * 60 thing as a 5minTTL as a global var in this test file?

File: integration/v3_lease_test.go [hunk 2]
```diff
@@ -366,6 +366,242 @@ func TestV3LeaseFailover(t *testing.T) {
...
+	if err != nil {
+		t.Fatal(err)
+	}
+	if lresp.Error != "" {
+		t.Fatal(lresp.Error)
+	}
+	if !leaseExist(t, clus, lresp.ID) {
```
Comment (xiang90) [inline integration/v3_lease_test.go]: remove this. if lease does not exit line 387 will fail.

File: integration/v3_lease_test.go [hunk 3]
```diff
@@ -366,6 +366,242 @@ func TestV3LeaseFailover(t *testing.T) {
...
+
+	// restart server and ensure lease still exists
+	clus.Members[0].Stop(t)
+	clus.Members[0].Restart(t)
+	clus.waitLeader(t, clus.Members)
+	if !leaseExist(t, clus, lresp.ID) {
+		t.Error("unexpected lease not exists")
+	}
+
+	nc, err := NewClientV3(clus.Members[0])
```
Comment (xiang90) [inline integration/v3_lease_test.go]: why do we need a new client?

File: integration/v3_lease_test.go [hunk 4]
```diff
@@ -366,6 +366,242 @@ func TestV3LeaseFailover(t *testing.T) {
...
+	lsc := toGRPC(clus.Client(0)).Lease
+
+	lresp, err := lsc.LeaseGrant(context.TODO(), &pb.LeaseGrantRequest{TTL: 5 * 60})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if lresp.Error != "" {
+		t.Fatal(lresp.Error)
+	}
+	if !leaseExist(t, clus, lresp.ID) {
```
Comment (xiang90) [inline integration/v3_lease_test.go]: remove this checking

File: integration/v3_lease_test.go [hunk 5]
```diff
@@ -366,6 +366,242 @@ func TestV3LeaseFailover(t *testing.T) {
...
```
Comment (xiang90) [inline integration/v3_lease_test.go]: remove this checking

**Comment by xiang90** [comment]
LGTM

**Comment by gyuho** [comment]
@xiang90 Without your `Fix Lease` PR, it fails:

```
go test -v -run TestV3LeaseRecoverKeyWithDetachedLease
=== RUN   TestV3LeaseRecoverKeyWithDetachedLease
--- FAIL: TestV3LeaseRecoverKeyWithDetachedLease (0.16s)
	v3_lease_test.go:530: only detached lease removed, key should remain
FAIL
exit status 1
FAIL	github.com/coreos/etcd/integration	0.169s
gyuho@gyuho: ~/go/src/github.com/coreos/etcd/integration $
go test -v -run TestV3LeaseRecoverKeyWithMutipleLease
=== RUN   TestV3LeaseRecoverKeyWithMutipleLease
--- FAIL: TestV3LeaseRecoverKeyWithMutipleLease (0.16s)
	v3_lease_test.go:589: only detached lease removed, key should remain
```

And we need to create a new client after restart because gRPC fails fast (we don't pass FailFast(false) in integration test). I added comments on this.

Will merge after greens, if there's no objection.

Thanks!

**Pull request merged**